src/components/functional/PaymentRequestTable/PaymentRequestTable.tsx

```diff
@@ -30,21 +37,72 @@ const PaymentRequestTable = ({ token, apiUrl = 'https://api.nofrixion.com/api/v1
     pageSize,
   );
 
+  const { deletePaymentRequest } = useDeletePaymentRequest(apiUrl, token);
+
   const localPaymentRequests: LocalPaymentRequest[] = paymentRequests.map((paymentRequest) =>
     RemotePaymentRequestToLocalPaymentRequest(paymentRequest),
   );
 
+  const onDeletePaymentRequest = async (paymentRequest: LocalPaymentRequest) => {
+    setPaymentRequestIdToDelete(paymentRequest.id);
+    setIsDeleteDialogOpen(true);
+  };
+
+  const onCopyPaymentRequestLink = async (paymentRequest: LocalPaymentRequest) => {
+    let link = `${apiUrl}/nextgen/pay/${paymentRequest.id}`;
+    await navigator.clipboard.writeText(link);
+
+    makeToast('success', 'Payment request link copied to clipboard.');
+  };
+
+  const closeDeleteDialog = () => {
+    setIsDeleteDialogOpen(false);
+  };
+
+  const confirmDeletePaymentRequest = async () => {
+    if (paymentRequestIdToDelete && paymentRequestIdToDelete !== '') {
+      let deleteError = await deletePaymentRequest(paymentRequestIdToDelete);
+
+      if (deleteError) {
+        makeToast('error', deleteError.title);
+        return;
+      }
+
+      makeToast('success', 'Payment request successfully deleted.');
+
+      await fetchPaymentRequests();
+      setPaymentRequestIdToDelete('');
+    }
+
+    setIsDeleteDialogOpen(false);
+  };
```

Comment:
I'd try to move as much logic from here to the `UI/PaymentRequestTable`. I just wonder if could be possible to manage to get the dialog in the UI component and then execute the request in here once we got the confirmation from wherever it is, Dialog, double click, etc...

Just to make the UI the most testable without the need of connecting to the API.